classes/local/results.php

```diff
@@ -523,9 +523,8 @@ public static function diary_print_user_entry($context, $course, $diary, $user,
             echo '<td style="width:35px;">'.get_string('entry', 'diary').':</td><td>';
             echo userdate($entry->timecreated);
             // 20201202 Added link to show all entries for a single user.
-            echo '  <a href="reportsingle.php?id='.$id
-                .'&user='.$user->id
-                .'&action=allentries">'.get_string('reportsingle', 'diary')
+            $url = new moodle_url('reportsingle.php', array('id' => $id, 'user' => $user->id, 'action' => 'allentries'));
```

Comment:
Try to remove array() as the default constructor for arrays, we should basically always be using [] except when matching with legacy code